src/giget.ts

```diff
@@ -102,20 +102,30 @@ export async function downloadTemplate(
   if (!options.offline) {
     await mkdir(dirname(tarPath), { recursive: true });
     const s = Date.now();
-    await download(template.tar, tarPath, {
-      headers: {
-        Authorization: options.auth ? `Bearer ${options.auth}` : undefined,
-        ...normalizeHeaders(template.headers),
-      },
-    }).catch((error) => {
-      if (!existsSync(tarPath)) {
-        throw error;
-      }
-      // Accept network errors if we have a cached version
-      debug("Download error. Using cached version:", error);
-      options.offline = true;
-    });
-    debug(`Downloaded ${template.tar} to ${tarPath} in ${Date.now() - s}ms`);
+
+    if (template.git) {
+      await cloneAndArchive(template.git, tarPath, { version: template.version });
```

Comment:
I think we should introduce a `download: () => {}` callback support for providers this way we can directly implement `git` as a provider without changing core.